src/app/packages/core/components/imagery-status/imagery-status.component.ts

```diff
@@ -38,29 +47,37 @@ export class ImageryStatusComponent implements OnInit {
 
 
 	favoriteOverlays: Overlay[];
-
+	isFavorite: boolean;
+	favoritesButtonText: string;
 
 	get description() {
-		return (this.overlay && this.overlay) ? new Date(this.overlay.photoTime).toUTCString() + ' - ' + this.overlay.sensorName : null;
+		return (this._overlay && this._overlay) ? new Date(this._overlay.photoTime).toUTCString() + ' - ' + this._overlay.sensorName : null;
 	}
 
 	get isNotGeoRegistered() {
-		return !MapFacadeService.isOverlayGeoRegistered(this.overlay);
+		return !MapFacadeService.isOverlayGeoRegistered(this._overlay);
 	}
 
 	constructor(protected store$: Store<any>, @Inject(CoreConfig) public coreConfig: ICoreConfig) {
 	}
 
-	isFavoriteOverlayDisplayed() {
-		return this.favoriteOverlays.some(o => o.id === this.overlay.id);
-	}
-
 	ngOnInit(): void {
-		this.favoriteOverlays$.subscribe((favoriteOverlays) => this.favoriteOverlays = favoriteOverlays);
+		this.favoriteOverlays$.subscribe((favoriteOverlays) => {
+			this.favoriteOverlays = favoriteOverlays;
+			this.updateFavoriteStatus();
+		});
 	}
 
 	toggleFavorite() {
-		this.store$.dispatch(new ToggleFavoriteAction(this.overlay));
+		this.store$.dispatch(new ToggleFavoriteAction(this._overlay));
+	}
+
+	updateFavoriteStatus() {
+		this.isFavorite = false;
+		if (this._overlay && this.favoriteOverlays && this.favoriteOverlays.length > 0) {
+			this.isFavorite = this.favoriteOverlays.some(o => o.id === this._overlay.id);
+		}
+		this.favoritesButtonText = this.isFavorite ? 'Remove from favorites' : 'Add to favorites';
```

Comment:
Why not just function `favoritesButtonText()` with `this.isFavorite ? 'Remove from favorites' : 'Add to favorites'
`instead of  `setter` , `getter`  and `onInit` ?